### PR TITLE
Use EdgeInsetsGeometry instead of EdgeInsets

### PR DESCRIPTION
**Pull request by orischwartz**
Should accommodate both `EdgeInsetsDirectional` and `EdgeInsets`.

## Description
Use the common superclass instead of just `EdgeInsets`

Before this PR, the following code will fail with `type 'EdgeInsetsDirectional' is not a subtype of type 'EdgeInsets'`:

```dart
  Widget build(BuildContext context) {
    return Container(
        child: Text("Hi"),
        padding: EdgeInsetsDirectional.only(start: 3),
        decoration: ShapeDecoration(
          shape: BorderDirectional(start: BorderSide(color: Colors.red, width: 3)),
        ));
  }
```

The getter delegates to 

https://github.com/flutter/flutter/blob/0b24a5a2ff9eccfba77bb68a0abcf1b7f0ae5b37/packages/flutter/lib/src/painting/borders.dart#L307

## Related Issues
Similar to #19305

### CODE REVIEW

**Review by goderbauer — COMMENTED** [review]
Thanks for the fix!

Can you please add a test to ensure that we never regress this and it continues to work? Thanks!

**Comment by goderbauer** [comment]
@orischwartz Are you planning to come back to this one to add a test?

**Comment by orischwartz** [comment]
@goderbauer Yes just haven't had a chance yet -- should be very straightforward to do.

**Comment by orischwartz** [comment]
Test added

File: packages/flutter/test/widgets/shape_decoration_test.dart [hunk 1]
```diff
@@ -61,6 +61,14 @@ Future<void> main() async {
     );
   }, skip: isBrowser);
 
+  test('ShapeDecoration with BorderDirectional', () {
+    var decoration = ShapeDecoration(
```
Comment (goderbauer) [inline packages/flutter/test/widgets/shape_decoration_test.dart]:
```suggestion
    final ShapeDecoration decoration = ShapeDecoration(
```

File: packages/flutter/test/widgets/shape_decoration_test.dart [hunk 2]
```diff
@@ -61,6 +61,14 @@ Future<void> main() async {
     );
   }, skip: isBrowser);
 
+  test('ShapeDecoration with BorderDirectional', () {
+    final ShapeDecoration decoration = ShapeDecoration(
```
Comment (Piinks) [inline packages/flutter/test/widgets/shape_decoration_test.dart]: The analyzer is failing here where `const` should be used.

**Review by Piinks — COMMENTED** [review]
Hi @orischwartz,
Welcome & thank you for the contribution! 🎉 
Please see below where the analyzer is failing.

**Review by Piinks — APPROVED** [review]
LGTM! 🎉 
We'll be able to merge this in once the build is green. Thanks again! :)

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**